HIP/hip-412.md

```diff
@@ -213,7 +233,149 @@ Note that mime types for directories are not uniformly defined. Some IPFS CIDs p
+
+The `files` array contains the main image hosted on IPFS, also including an optional checksum for validation purposes. The `image` field serves as a preview image for NFT tooling where a programatically generated image doesn't work or you would like to have a different preview image. Therefore, the `image` field remains optional.
+
+Further, no additional properties are defined on the JSON object on root level. Any additional properties you want to define should go into the `properties` object.
+
+You can optionally define `attributes` to calculate rarity scores. We've added the field `display_type` similar to the OpenSea standard which defines how the attribute should be displayed.
+
+The standard also allows for localization. Each locale links to another metadata file that contains the localized metadata and files. This allows for a clean metadata structure. Don't define another localization object for a localized metadata file to avoid infinite looping when parsing an NFT's metadata file.
+
+**A JSON schema can be found [here](https://gist.github.com/michielmulders/1565e7759d692b65fa8c38cfe15b010c).**
+
+```json
+{
+	"name": "Example NFT 001",
+	"creator": "Jane Doe, John Doe",
+	"creatorDID": "did:hedera:mainnet:7Prd74ry1Uct87nZqL3ny7aR7Cg46JamVbJgk8azVgUm;hedera:mainnet:fid=0.0.123",
+	"description": "This describes my NFT",
+	"image": "https://myserver.com/nft-001.png",
+	"sha256_checksum": "ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad",
+	"type": "image/png",
+	"format": "HIP412",
+	"properties" : {
+		"external_url": "https://nft.com/mycollection/001"
+	},
+	"files": [
+		{
+			"uri": "ipfs://bawlkjaklfjoiaefklankfldanmfoieiajfl",
+			"sha256_checksum": "ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad",
+			"is_default_file": true,
+			"type": "image/png"
+		},
+		{
+			"uri": "ipfs://yusopwpksaioposjfopiapnnjlsl",
+			"type": "image/png"
+		}
+	],
+	"attributes": [
+		{
+			"trait_type": "colour",
+			"value": "red"
+		},
+		{
+			"trait_type": "hasPipe",
+			"value": true
+		},
+		{
+			"trait_type": "coolness",
+			"display_type": "boost",
+			"value": 10,
+			"max_value": 100
+		},
+		{
+			"trait_type": "stamina",
+			"display_type": "percentage",
+			"value": 83
+		}
+		,
+		{
+			"trait_type": "birth",
+			"display_type": "datetime",
+			"value": "23-03-1993"
+		}
+	],
+	"localization": [
+		{
+			"locale": "en",
+			"is_default_locale": true
+		},
+		{
+			"locale": "es",
+			"metadata_uri": "ipfs://keiyidooajiaefklankfldanmfoieoakddqtyty"
+		},
+		{
+			"locale": "jp",
+			"metadata_uri": "ipfs://yidooajiaefiakfldanm12554woakoakga4sfda"
+		}
+	]
+}
+```
+
+Here's a clarification for the most important fields for the `HIP412` format.
+
+#### files
+
+(Required) `files` array holds one or many files. Because the optional `image` property can be used as a preview image, the `files` array should at least hold one `file` object that represents the NFT. 
```

Comment:
Well, this is a point of discussion, we have two options:

1. Use the `image` field as an optional preview image field if you don't want NFT tooling to programmatically generate one. The `files` array is then used as the main place to store the NFTs files. 

2. Use the `image` field as either an optional preview image field or a way to store the NFT's image. Then any additional files go into the `files` array which makes the `files` array optional. 

Wouldn't it be more straightforward to have option 1? Only use `image` as a preview image and all other files in the `files` array. This does avoid guesswork for NFT tooling to which file is the correct file for an NFT.

Reply:
Since other chains are already storing the main image in the `image` field, I think option 2 provides the best cross platform solution and also simplest to explain.

We need to have a way to represent all NFTs visually for display in wallets and on marketplaces whether they are PFP, music, 3D animations, domain names, house deeds, carbon credits etc. The `image` field is perfectly suited for this.

In reality, marketplaces/wallets are all going to be caching and/or generating a preview version of the `image` field since they can't guarantee that those images are all going to be optimised for fast loading.

I would advocate that `image` be a required field that you can store whatever size image you like in but that we recommend using as a preview image (350px to 1080px as already mentioned) and then storing high-res versions or other file types in the files array.

Then we can boil it down to this:
> The `image` will be displayed in wallets and marketplaces by default. Some platforms may support displaying other file types such as 3D files, audio or video. Use the `is_default_file` in your `files` array to mark that file as the one you would prefer to have displayed by default if the platform supports it.